src/lib/Microsoft.Health.Fhir.Ingest.Template/HandlerProxyTemplateFactory.cs

```diff
@@ -12,8 +14,29 @@ public abstract class HandlerProxyTemplateFactory<TInput, TOutput> :
         IResponsibilityHandler<TInput, TOutput>
         where TOutput : class
     {
+        private readonly IList<string> _serializationErrors = new List<string>();
+
+        protected IList<string> SerializationErrors
+        {
+            get { return _serializationErrors; }
+        }
+
         public abstract TOutput Create(TInput jsonTemplate);
 
         TOutput IResponsibilityHandler<TInput, TOutput>.Evaluate(TInput request) => Create(request);
+
+        protected JsonSerializer GetJsonSerializer()

```

Comment:
What are the reasons for keeping the custom serializer settings? Are the seriailzation errors thrown not enough?

I had written a prior comment but looks like I forgot to publish it.  I am concerned with using a delegate approach for the error handler.  This can lead to memory leaks.  A handler like this is an object reference that can keep the object reference alive and prevents garbage collection.

Reply:
Discussed offline. This has been addressed.